core/src/main/java/io/questdb/client/Sender.java

```diff
@@ -430,6 +669,175 @@ public LineSenderBuilder port(int port) {
             return this;
         }
 
+        /**
+         * Configures the maximum time the Sender will spend retrying upon receiving a recoverable error from the server.
+         * <br>
+         * This setting is applicable only when communicating over the HTTP transport, and it is illegal to invoke this
+         * method when communicating over the TCP transport.
+         * <p>
+         * Recoverable errors are those not caused by the client sending invalid data to the server. For instance,
+         * connection issues or server outages are considered recoverable errors, whereas attempts to send a row
+         * with an incorrect data type are not.
+         * <p>
+         * Setting this value to zero disables retries entirely. In such cases, the Sender will throw an exception
+         * immediately. It's important to note that the Sender does not retry operations that fail
+         * during {@link #close()}. Therefore, it is recommended to explicitly call {@link #flush()} before closing
+         * the Sender.
+         * <p>
+         * <b>Warning:</b> Retrying may lead to data duplication. It is advisable to use
+         * <a href="https://questdb.io/docs/concept/deduplication/">QuestDB deduplication</a> to mitigate this risk.
+         * <p>
+         * Default value: 10,000 milliseconds.
+         *
+         * @param retryTimeoutMillis the maximum retry duration in milliseconds.
+         * @return this instance, enabling method chaining.
+         */
+        public LineSenderBuilder retryTimeoutMillis(int retryTimeoutMillis) {
+            if (this.retryTimeoutMillis != MAX_RETRY_MILLIS_DEFAULT) {
+                throw new LineSenderException("max retries was already configured ")
+                        .put("[retry-timeout-millis=").put(this.retryTimeoutMillis).put("]");
+            }
+            if (retryTimeoutMillis < 0) {
+                throw new LineSenderException("retry timeout cannot be negative ")
+                        .put("[retry-timeout-millis=").put(retryTimeoutMillis).put("]");
+            }
+            this.retryTimeoutMillis = retryTimeoutMillis;
+            return this;
+        }
+
+        /**
+         * Configures a QuestDB server address. The address must be a full URL, including a protocol, host and port.
+         * This is the preferred way of configuring a Sender.
+         * <p>
+         * Example 1: http://localhost:9000 - this will configure a Sender to communicate over HTTP transport
+         * <br>
+         * Example 2: tcp://localhost:9009 - this will configure a Sender to communicate over TCP transport
+         *
+         * @param url a full URL to a QuestDB server.
+         * @return this instance for method chaining
+         */
+        public LineSenderBuilder url(CharSequence url) {
```

Comment:
`username:password@host...` and `token@host...` syntax seems to be undocumented. Is this on purpose?

Reply:
resolving as URL support was removed completely